src/FamilyTree.jsx

```diff
@@ -0,0 +1,179 @@
+import React from 'react';
+import BitByteLabel from './BitByteLabel.jsx';
+import Tree from 'react-d3-tree';
+import './FamilyTree.css';
+
+/**
+ * Our custom definition of a label on the tree 
+ * This definition will allow us to view the information about a 
+ * bit / byte by hovering over their name on the tree
+ */
+class NodeLabel extends React.PureComponent{
+  
+  /* Constructor to initialize state and bind function */
+  constructor(props){
+    super(props);
+    this.handleMouseHover = this.handleMouseHover.bind(this);
+    this.state= {
+      isHovering:false
+    }
+  }
+
+  /* Event handler */
+  handleMouseHover() {
+    this.setState(this.toggleHoverState);
+  }
+
+  /* Toggles the isHovering state */
+  toggleHoverState(state){
+    return{
+      isHovering: !state.isHovering,
+    };
+  }
+
+  /* Returns the actual label*/
+  render(){
+    const{className, nodeData} = this.props
+    return(
+      // This division will react to mouse hovering
+      <div 
+        className = {className} 
+        onMouseEnter = {this.handleMouseHover}
+        onMouseLeave = {this.handleMouseHover}    
+      >
+        <p className="label-text">{nodeData.attributes.node_name}</p>
+        {
+          this.state.isHovering && // The below info will show up on hover
+          <div class = "PopUp">
+          <BitByteLabel
+            id = {nodeData.attributes.id}
+            node_name = {nodeData.attributes.node_name}
+            tree_name = {nodeData.attributes.tree_name}
+            quarter_joined = {nodeData.attributes.quarter_joined}
+            college = {nodeData.attributes.college}
+            major = {nodeData.attributes.major}
+            class_year = {nodeData.attributes.class_year}
+            linkedin = {nodeData.attributes.linkedin}
+            facebook = {nodeData.attributes.facebook}
+            instagram = {nodeData.attributes.instagram}
+          />
+          </div>
+        }
+      </div>
+    )
+  }
+}
+
+/* Hard-coded data below */
+const bitByteTreeData = [
```

Comment:
For the future we should probably store hard-coded data in a separate json and import it rather than putting it directly in the FamilyTree.jsx

Reply:
Ok, did that in the most recent push